fast64_internal/f3d/f3d_gbi.py

```diff
@@ -1426,10 +1426,49 @@ def __init__(self, texCSeparate, savePNG, includeDir, exportPath = ''):
 		self.includeDir = includeDir
 		self.exportPath = exportPath
 
+#  SetTileSize Scroll Data
+class FSetTileSizeScrollField:
+	def __init__(self):
+		self.s = 0
+		self.t = 0
+
+def tile_func(direction: str, speed: int, cmdNum: int):
+	if speed == 0 or speed is None:
+		return None
+
+	func = f'shift_{direction}'
+
+	if speed < 0:
+		func += '_down'
+
+	return f'\t{func}(mat, {cmdNum}, PACK_TILESIZE(0, {abs(speed)}));'
+
+def mat_tile_scroll(
+	mat: str,
+	tex0: FSetTileSizeScrollField,
+	tex1: FSetTileSizeScrollField,
+	cmdNum0: int,
+	cmdNum1: int
+):
+	func = f'void scroll_sts_{mat}()'
```

Comment:
could you explain what this PR achieves, I seem to understand it writes C functions, are those to be called manually then?

(I'm worried that undocumented features may just never get used)

Reply:
it writes C functions and those end up getting called with the remainder of the texture scrolling functions. all texture scrolling functions in general get called from the main game thread while the game is active and unpaused.

Quite a few people are planning on using this already haha, its pretty important as far as scrolling textures go. It modifies the S and T values of the SetTileSize command for either tex0 or tex1, this way you can have only one texture in a multitexture scrolling, or have the two textures scrolling in different directions. Like the wall in Zora's Domain.

It'd be really helpful in OOT as well, but the implementation would be much more different.